pkg/util/winutil/iisconfig/apmtags_test.go

```diff
@@ -0,0 +1,117 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+//go:build windows
+
+package iisconfig
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestAppConfig(t *testing.T) {
+	path, err := os.Getwd()
+	require.Nil(t, err)
+
+	apppath := filepath.Join(path, "testdata", "app_1.config.xml")
+	//apppath := filepath.Join(path, "testdata", "iisconfig.xml")
+	iisCfg, err := ReadDotNetConfig(apppath)
+	assert.Nil(t, err)
+	assert.NotNil(t, iisCfg)
+
+	assert.Equal(t, iisCfg.DDService, "service1")
+	assert.Equal(t, iisCfg.DDEnv, "false")
+	assert.Equal(t, iisCfg.DDVersion, "1.0-prerelease")
+
+}
+
+func TestPathSplitting(t *testing.T) {
+
+	t.Run("Test Root path", func(t *testing.T) {
+		sp := splitPaths("/")
+		assert.Equal(t, 0, len(sp))
+	})
+
+	t.Run("Test path depth 3", func(t *testing.T) {
+		sp := splitPaths("/path/to/app")
+		assert.Equal(t, 3, len(sp))
+		assert.Equal(t, "path", sp[0])
+		assert.Equal(t, "to", sp[1])
+		assert.Equal(t, "app", sp[2])
+	})
+
+	t.Run("Test path depth 3 with trailing slash", func(t *testing.T) {
+		sp := splitPaths("/path/to/app/")
+		assert.Equal(t, 3, len(sp))
+		assert.Equal(t, "path", sp[0])
+		assert.Equal(t, "to", sp[1])
+		assert.Equal(t, "app", sp[2])
+	})
+
```

Comment:
I also recommend testing multiple separators: `/foo///` and empty ``

Reply:
These are URL paths. are those valid URL paths?
(btw, the fact that it's URL paths is why I had to write my own `splitPaths`.
URL path separators are expected to be `/`; the go path split since we're running on windows is expecting the `\` separators. .  The implementation of `splitPaths` also calls `clean` to handle the rest.